src/mlpack/methods/ann/layer/radial_basis_function_impl.hpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file radial_basis_impl.hpp
+ * @author Himanshu Pathak
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_RBF_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_RBF_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "dropout.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+RBF<InputDataType, OutputDataType>::RBF() :
+    inSize(0),
+    outSize(0),
+    reset(false)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+RBF<InputDataType, OutputDataType>::RBF(
+    const size_t inSize,
+    const size_t outSize) :
+    inSize(inSize),
+    outSize(outSize),
+    reset(false)
+{
+  weights.set_size(outSize * inSize + outSize, 1);
+}
+
+template<typename InputDataType, typename OutputDataType>
+void RBF<InputDataType, OutputDataType>::Reset()
+{
+  centres = arma::mat(weights.memptr(), inSize, outSize, false, false);
+  sigmas = arma::mat(weights.memptr() + centres.n_elem,
+      1, outSize, false, false);
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename eT>
+void RBF<InputDataType, OutputDataType>::Forward(
+    const arma::Mat<eT>& input,
+    arma::Mat<eT>& output)
+{
+  if(!reset)
+  {
+    centres = arma::normcdf(centres, 0,1);
+    sigmas.ones();
+    sigmas = sigmas / outSize;
+    reset = true;
+  }
+  distances = arma::mat(outSize, input.n_cols);
+
+  for (size_t i = 0; i < input.n_cols; i++)
+  {
+    arma::mat temp = centres.each_col() - input.col(i);
+    distances.col(i) = arma::pow(arma::sum(
+                                 arma::pow((temp),
+                                 2), 0), 0.5).t();
```

Comment:
@saksham189 When I was doing sum along rows and using c,x variable which is a wrong implementation it was passing now it is failing??

Reply:
You are not training the centres and scaling factors. For that you would not need to add a gradient method as shown here: https://github.com/mlpack/mlpack/blob/4c1389fa32c7021b019c42399a744b3defe0c464/src/mlpack/methods/ann/layer/linear_impl.hpp#L75

Reply:
I was confused about this. I read a paper Gradient assisted RBFN. So I thought in simple RBFN we don't need to find a gradient for the centre. I will implement it.